src/components/app-switcher/app-switcher.js

```diff
@@ -0,0 +1,164 @@
+         * @type {String}
+         * @default ''
+         */
+    applicationJson: PropTypes.string.isRequired,
+
+        /**
+         * The title of the menu item
+         *
+         * @property menuTitle
+         * @type {String}
+         * @default ''
+         */
+    menuTitle: PropTypes.string
+
+  }
+
+    /**
+     * Pseudo private class for the styles
+     *
+     * @method styles
+     * @private
+     * @return Hash
+     */
+
+  styles = () => {
+    return {
+      appItem: "carbon-switcher-app-item",
+      menu: classNames((this.state.active ? "carbon-switcher-menu-active" : ""), "carbon-switcher-menu-title"),
+      sectionTitle: "carbon-switcher-section-title",
+      drawer: "carbon-switcher-drop-down",
+      arrowUp: "carbon-switcher-arrow-up"
+    };
+  }
+
+    /**
+     * Constructor method setting properties
+     *
+     * @method constructor
+     * @return {Void}
+     */
+
+  constructor(...props) {
+    super(...props);
+
+    this.state = {
+      active: false
+    };
+  }
+
+    /**
+     * Handle when the user clicks on the main menu icon
+     *
+     * @method handleOpenWindow
+     * @return {Void}
+     */
+
+  handleOpenWindow = () => {
+    document.addEventListener('click', this.handleCloseWindow);
+    this.setState({active: true});
+  }
+
+    /**
+     * Handle when the user clicks anywhere on the screen after
+     * the main menu is activated
+     *
+     * @method handleCloseWindow
+     * @return {Void}
+     */
+
+  handleCloseWindow = () => {
+    document.removeEventListener('click', this.handleCloseWindow);
+    this.setState({active: false});
+  }
+
+    /**
+     * Compose each of the individual pieces from applicationJson
+     *
+     * @method composeDrawer
+     * @return JSX Elements
+     */
+
+  compose = () => {
+    let parsedJson = JSON.parse(this.props.applicationJson) || {};
+
+    return parsedJson.items.map((section, index) => {
+      let items = section.items.map((item,index) => {
+        return (<DrawerItem href = {item.href} key = {'index-' + index} cname = {this.styles().appItem} name={item.name} /> );
+      });
+
+      return (
+          <DrawerSection cname = {this.styles().sectionTitle} key={'index-' + index} items = {items} title={section.title}/>
+        );
+    });
+
+  };
+
+
+    /**
+     * The main render method
+     *
+     * @method render
+     * @return JSX Elements
+     */
+
+  render() {
+    return (
+            <span className={this.styles().menu} onClick={this.handleOpenWindow}>
+              <Icon type='grid' />
+              {this.props.menuTitle}
+              { this.state.active ?
+              <div onClick={this.handleCloseWindow} className={this.styles().drawer}>
+                        {this.compose()}
+                        <div className={this.styles().arrowUp} />
+                     </div> : null }
+                </span>
+    );
+  }
+
+}
+
+class DrawerSection extends React.Component {
```

Comment:
These two classes should be [functional components](https://facebook.github.io/react/docs/components-and-props.html).

This would be re-written as:

```
const DrawerSection = props => {
  <div>
    <div className = {props.cname} > {props.title} </div>
    <div>{props.items}</div>
  </div>
};
```

This applies to both